crates/bevy_reflect/src/serde/de.rs

```diff
@@ -378,6 +400,11 @@ impl<'a, 'de> DeserializeSeed<'de> for TypedReflectDeserializer<'a> {
             return Ok(value);
         }
 
+        if let Some(deserialize_reflect) = self.registration.data::<ReflectDeserializeReflect>() {
```

Comment:
As discussed on Discord, the naming is really not ideal. The variable here `deserialize_reflect` even ends up being the same as the one in the other block above, despite the types being different.

I proposed using `Registry` somewhere since that data/trait inherently adds a dependency to the type registry. So maybe `ReflectDeserializeRegistry`? (with `Registry` as suffix like `Seed` is, since the registry value kind of replace the seed in `DeserializeSeed`)

Reply:
Agreed. I think `DeserializeRegistry` (`ReflectDeserializeRegistry`) is pretty good. 

A few others:
- `RegistryDeserialize` (`ReflectRegistryDeserialize`)
- `DeserializeTypeRegistry` (`ReflectDeserializeTypeRegistry`)
- `DeserializeReflected` (`ReflectDeserializeReflected`), though, maybe that one's still not different enough

Does anyone else have any other thoughts?

Reply:
Hm, I think I like the `WithRegistry` suffix the best. I'll switch it over to that for now, unless anyone else has any other opinions.